fog/view/server/src/block_tracker.rs

```diff
@@ -461,104 +434,30 @@ mod tests {
         let mut block_tracker = BlockTracker::new(logger.clone());
 
         assert_eq!(
-            block_tracker.highest_fully_processed_block_count(&[], &[]),
-            0
-        );
-    }
-
-    // A missing range that doesn't start at block 0 should not affect the count.
-    #[test_with_logger]
-    fn highest_fully_processed_block_count_some_missing_blocks(logger: Logger) {
-        let mut block_tracker = BlockTracker::new(logger.clone());
-
-        assert_eq!(
-            block_tracker.highest_fully_processed_block_count(&[], &[BlockRange::new(1, 10)]),
-            0
-        );
-    }
-
-    // A missing range that does start at block 0 should advance the count to the end of the
-    // range.
-    #[test_with_logger]
-    fn highest_fully_processed_block_count_starts_with_missing_blocks(logger: Logger) {
-        let mut block_tracker = BlockTracker::new(logger.clone());
-
-        assert_eq!(
-            block_tracker.highest_fully_processed_block_count(&[], &[BlockRange::new(0, 10)]),
-            10
+            block_tracker.highest_fully_processed_block_count(&[]),
+            (0, None)
         );
     }
 
-    // Multiple missing ranges are handled appropriately
-    #[test_with_logger]
-    fn highest_fully_processed_block_missing_blocks_consecutivie_missing1(logger: Logger) {
```

Comment:
This test relates to my comment above with how the view server advances to the next block. I think if the ingress keys are not all consecutive and somehow there is a gap then the view server will stop progressing (unless I misunderstood something).